kiteconnect/connect.py

```diff
@@ -821,13 +824,27 @@ def _parse_mf_instruments(self, data):
             row["last_price"] = float(row["last_price"])
 
             # Parse date
-            if len(row["last_price_date"]) == 10:
-                row["last_price_date"] = dateutil.parser.parse(row["last_price_date"]).date()
+            if self.is_timestamp(row["last_price_date"]):
+                row["last_price_date"] = self.set_tz(row["last_price_date"]).date()
 
             records.append(row)
 
         return records
 
+    def is_timestamp(self, string):
+        """Checks if string is timestamp"""
+        try:
+            parse(string)
+            return True
+        except ValueError:
+            return False
+
+    def set_tz(self, string):
```

Comment:
May be you can call this as `parseDateTime`? because since this basically creating a Python datetime object from string with timezone set.

Reply:
Done.